File: src/main/java/work/lclpnet/ap2/game/pillar_battle/item/SingletonItemClass.java
package work.lclpnet.ap2.game.pillar_battle.item;

import net.minecraft.item.Item;
import net.minecraft.item.ItemStack;

import java.util.Random;

public record SingletonItemClass(Item item) implements ItemClass {
    @Override
    public ItemStack getRandomStack(Random random) {
        return new ItemStack(item);
    }
}

File: src/minigames/pillar_battle/src/main/kotlin/work/lclpnet/ap2/game/pillar_battle/item/SingletonItemClass.kt
package work.lclpnet.ap2.game.pillar_battle.item

import net.minecraft.world.item.Item
import net.minecraft.world.item.ItemStack
import java.util.*
import java.util.stream.Stream

data class SingletonItemClass(val item: Item) : ItemClass {
    override fun getRandomStack(random: Random): ItemStack = ItemStack(item)
    override fun stream(): Stream<Item> = Stream.of(item)
}
